ballerina/hpke.bal

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2024 WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Represents the supported HPKE algorithms.
+public type HpkeAlgorithm KYBER768|RSA_KYBER768;
+
+# Represent the supported symmetric key sizes for AES algorithm.
+public type AesKeySize 16|24|32;
+
+# Represents the encapsulated secret and the ciphertext used in Hybrid Public Key Encryption (HPKE).
+#
+# + algorithm - The hybrid public key encryption algorithm used
+# + encapsulatedSecret - The encapsulated secret
+# + cipherText - The encrypted data
+public type HybridEncryptionResult record {|
+    HpkeAlgorithm algorithm;
```

Comment:
Do we need to specify the algorithm again in the result as well? The user is aware of the algorithm when they call the function.